### DISCUSSION TITLE
Cross-platform plans?

### DISCUSSION THREAD

**Discussion by tovrstra** [absperf/asyncinotify#10] [Q&A]
I was wondering if there are any plans to extend asyncinotify to other platforms? I'm currently using watchdog, but I'm hitting [one of its long-standing issues](https://github.com/gorakhargosh/watchdog/issues/275). While looking for an alternative, asyncinotify looks really nice and first tests are promising. I even appreciate the absence of the recursive option, which is a source of trouble in other implementations. My only hesitation is that asyncinotify would be Linux-only dependency. (I haven't found anything comparable to asyncinotify for FSEvents on macos or for Windows.)

**Comment by TaylorRichberger** [comment]
Nope, no plans at all, and I would reject any attempt to add any other platforms to this library (edit: except other platforms that support the same inotify API, like FreeBSD), because it's only intended to be inotify.

I'd think that any cross-platform library should depend upon platform-specific ones under the covers rather than do all the implementation directly, so things can be more focused and easier to debug.  Either that, or something akin DB-API 2.0 for asynchronous filesystem events, where you have several libraries that conform to a common specified interface for filesystem events so they can be used largely interchangeably.  If we had something like that, I'd happily write an interface for this library to conform to it.

That said, I'd gladly help out with a wrapping library, at least as far as the inotify bits are concerned.  I have no interest in touching the Windows or MacOS equivalents myself, but I'm more than happy to help with anybody else's efforts to do so.
